src/main/java/controllers/FeedPresenter.java

```diff
@@ -21,6 +22,22 @@ public FeedPresenter(InOut inOut, User currentUser, Feed currentUsersFeed) {
 
     public void run(ShellAction action) {
         // TODO: get user inputs/display info based on the ShellAction
+        if (action == ShellAction.BROWSEFEED) {
+            // Asks the user for a filter (can be none)
+            String filterInput = this.getFilterInput();
+
+            if (!filterInput.equals("none")) {
+                // Calls FeedManager to update the feed to show only the filtered posts
+                this.setFeedFilter(filterInput);
+            }
+
+            // Gets current user's feed
+            Feed currentFeed = this.feedManager.getCurrentUsersFeed();
+            // Convert the feed into strings
+            ArrayList<ArrayList<String>> feedInStrings = this.convertFeedToString(currentFeed);
+            // Send the strings to InOut to be shown
+            this.setOutputToInOut(feedInStrings);
+        }
     }
 
     public void setCurrentUser() {}
```

Comment:
I think other than the run method all the other methods can be private. Do you think this makes sense or are there some methods that need to be public?